File: scheduling/scheduling-annotation-processor/src/test/java/ru/tinkoff/kora/scheduling/annotation/processor/controller/ScheduledWithCron.java
package ru.tinkoff.kora.scheduling.annotation.processor.controller;

import org.quartz.JobExecutionContext;
import ru.tinkoff.kora.scheduling.quartz.ScheduleWithCron;

public class ScheduledWithCron {
    @ScheduleWithCron("i can't cron")
    public void noArgs() {}

    @ScheduleWithCron("i can't cron")
    public void withCtx(JobExecutionContext jobExecutionContext) {}

    @ScheduleWithCron(value = "i can't cron", identity = "someIdentity")
    public void withIdentity() {}

    @ScheduleWithCron(value = "i can't cron", config = "some config")
    public void withConfig() {}

}

File: scheduling/scheduling-ksp/src/test/kotlin/ru/tinkoff/kora/scheduling/ksp/controller/ScheduledWithCron.kt
package ru.tinkoff.kora.scheduling.ksp.controller

import org.quartz.JobExecutionContext
import ru.tinkoff.kora.scheduling.quartz.ScheduleWithCron

class ScheduledWithCron {
    @ScheduleWithCron("i can't cron")
    fun noArgs() {
    }

    @ScheduleWithCron("i can't cron")
    fun withCtx(jobExecutionContext: JobExecutionContext?) {
    }

    @ScheduleWithCron(value = "i can't cron", identity = "someIdentity")
    fun withIdentity() {
    }

    @ScheduleWithCron(value = "i can't cron", config = "some config")
    fun withConfig() {
    }
}
